lib/API/JSON/API.php

```diff
@@ -79,7 +79,11 @@ function init() {
 
   function setupAjax() {
     $this->wp->doAction('mailpoet_api_setup', [$this]);
-    $this->setRequestData($_POST);
+    if (isset($_POST['api_version'])) {
+      $this->setRequestData($_POST);
+    } else {
+      $this->setRequestData($_GET);
+    }
```

Comment:
Allowing GET for all API endpoints is probably not a good idea. GET can transport a limited amount of data, GET should have no side-effects (well, in REST), GET can be cached, and it could cause security issues (pointing an admin to URL that would perform something).

What we can do is to go a bit forward REST here maybe. Converting this one single endpoint to `GET`, completely (allowing only GET for this one). Another option is to add a GET wrapper - a new route just for this. There is also https://developer.wordpress.org/rest-api/ but that may be a bigger topic, not sure.